google/cloud/spanner/partition_options.h

```diff
@@ -54,11 +54,20 @@ struct PartitionOptions {
    * returned may be smaller or larger than this maximum count request.
    */
   absl::optional<std::int64_t> max_partitions;
+
+  /**
```

Comment:
Are we planning to deprecate these `*Options` classes?  I don't mean remove, I mean just stop recommending them.  If so, we should stop adding features to them.  Probably a topic for future discussion.

Reply:
We have kind-of deprecated them in the `Client` API, in as much as those interfaces are marked as only for backwards compatibility, and we encouraged users to migrate away in a changelog note.

However, they are still used in the `Connection` API, so they're not really deprecated individually, and we still depend on them being able to carry new data.  I don't believe we have any plans to change that.